src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsReadStream.cs

```diff
@@ -0,0 +1,341 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+
+namespace Microsoft.Data.SqlClientX.IO
+{
+    /// <summary>
+    /// A stream which handles reading TDS messages from the wire, and extracting the data from the 
+    /// packets in the message packets.
+    /// </summary>
+    internal class TdsReadStream : Stream, ITdsReadStream
+    {
+        #region Private Fields
+        
+        private Stream _underlyingStream;
+
+        // The buffer to hold the TDS read data.
+        private byte[] _readBuffer;
+
+        // The read pointer inside the buffer.
+        private int _readIndex { get; set; } = 0;
```

Comment:
Do you mean for these to be fields? They're properties at the moment. `TdsWriteStream` has the same pattern.

Reply:
I don't and thanks for pointing this out. I will make them fields. 

_spid will be a field which will back a property that I will introduce.